File: leetcode/0003-longest-substring-without-repeating-characters/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        max_len = 0
        for i in range(len(s)): 
            seen = set()
            for j in range (i, len(s)):
                if s[j] in seen:
                    break
                else:
                    seen.add(s[j])
            max_len = max(max_len, len(seen))
    
        return max_len

File: leetcode/0003-longest-substring-without-repeating-characters/solution.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        
        List<string> substrings = new List<string>();
        string substring = "";
        int substringLength = 0;

        for (int i=0; i < s.Length; i++) {
            if (!substring.Contains(s[i]) ){
                substring += s[i];
            } else {
                int iOfDup = substring.IndexOf(s[i]);
                substring = substring.Remove(0, iOfDup + 1);
                substring += s[i];
            }
            if (substringLength < substring.Length)
                substringLength = substring.Length;
        }
        return substringLength;
    }
}
